querybook/server/logic/vector_store.py

```diff
@@ -114,46 +140,92 @@ def record_table(
         if get_vector_store().should_skip_table(table):
             return
 
-        summary = ai_assistant.summarize_table(table_id=table.id, session=session)
-
+        metastore_id = table.data_schema.metastore_id
         full_table_name = f"{table.data_schema.name}.{table.name}"
+
+        sample_query_cells = get_sample_query_cells_by_table_name(full_table_name)
+
+        # ingest table summary
+        summary = ai_assistant.summarize_table(
+            metastore_id=metastore_id,
+            table_name=full_table_name,
+            sample_queries=[q["query_text"] for q in sample_query_cells[:5]],
+            session=session,
+        )
+
         metadata = {
             "type": "table",
             "tables": [full_table_name],
-            "table_id": table.id,
+            "table_ids": [table.id],
+            "metastore_id": metastore_id,
         }
         doc_id = _get_table_doc_id(table.id)
         create_and_store_document(summary, metadata, doc_id)
+
+        # ingest sample queries summary
+        if ingest_sample_queries:
+            for query_cell in sample_query_cells:
+                LOG.info(f"Ingesting sample query cell: {query_cell}")
+                record_query_cell_from_es(es_query_cell=query_cell, session=session)
```

Comment:
if the query cell is referenced by 2 tables, would this cause duplicate ingestion?